### PR TITLE
System package quoting fixes

### PR DESCRIPTION
**Pull request by mkoeppe**
The format of the `distros/*.txt` was underspecified. We now clarify that there should be 1 package per line and no shell quoting. 

We extend the system package scripts so that they apply shell quoting when they print shell commands.
This fixes a failure seen in CI due to underquoting: https://github.com/sagemath/sage/actions/runs/7202678226/job/19621287638#step:11:134

We also wrap the command lines that are displayed by `sage-spkg-info` and formatted for the installation guide.
https://pianomister.github.io/diffsite/?url1=https%3A%2F%2Fdeploy-preview-36910--sagemath-tobias.netlify.app%2Fhtml%2Fen%2Finstallation%2Fsource&url2=https%3A%2F%2Fsagemath-tobias.netlify.app%2Fhtml%2Fen%2Finstallation%2Fsource

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: build/pkgs/_bootstrap/distros/fedora.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Packages needed for ./bootstrap
-autoconf automake libtool
+autoconf
+automake
+libtool
 # Fedora 26 needs:
```
Comment (dimpase) [inline build/pkgs/_bootstrap/distros/fedora.txt]: do we need this line?

Comment (mkoeppe) [reply build/pkgs/_bootstrap/distros/fedora.txt]: I'll remove this comment

File: build/pkgs/gfortran/distros/macports.txt [hunk 2]
```diff
@@ -1 +1,2 @@
-gcc10 +gfortran
+gcc10
++gfortran
```
Comment (dimpase) [inline build/pkgs/gfortran/distros/macports.txt:2]: What's that `+` there?

File: build/pkgs/mathjax/distros/gentoo.txt [hunk 3]
```diff
@@ -1 +1 @@
-">=dev-libs/mathjax-3"
+>=dev-libs/mathjax-3
```
Comment (dimpase) [inline build/pkgs/mathjax/distros/gentoo.txt:1]: `>=` ?

File: build/pkgs/gfortran/distros/macports.txt (hunk 2 above)
Comment (mkoeppe) [reply build/pkgs/gfortran/distros/macports.txt:2]: I don't remember, but I'm pretty sure it's not a typo. We can revisit this if we ever get #35667 unstuck

File: build/pkgs/mathjax/distros/gentoo.txt (hunk 3 above)
Comment (mkoeppe) [reply build/pkgs/mathjax/distros/gentoo.txt:1]: What about it?

Comment (dimpase) [reply build/pkgs/mathjax/distros/gentoo.txt:1]: Oops, sorry. It's OK. I never saw such syntax, but it's valid

**Comment by dimpase** [comment]
OK

**Review by dimpase — APPROVED** [review]

File: build/pkgs/_bootstrap/distros/fedora.txt (hunk 1 above)
Comment (mkoeppe) [reply build/pkgs/_bootstrap/distros/fedora.txt]: Done in 7c0eb333eaf

**Comment by mkoeppe** [comment]
Thanks!

**Comment by mkoeppe** [comment]
Marking it blocker so that blocker #36840 does not break the portability CI.

**Pull request merged**